frontend/src/components/filter/FilterMenu.js

```diff
@@ -161,20 +178,31 @@ export default function FilterMenu({ filters, onApplyFilters }) {
     setItems(newItems);
   };
 
+  const clearAllFilters = () => {
+    // this looks a little strange, right?
+    // well, we don't want to clear out things like the region, just the filters that can be set
+    // in the UI
+    const newItems = items.filter((item) => prohibitedFilters.includes(item.topic));
```

Comment:
If this component is only passed filters that can be changed in the UI (in the grantee page the `granteeId` filter isn't passed to this component) does it need to worry about not trampling "static" filters? If I'm mistaken can we introduce some concept of "static" filters that the `filter` components don't even see. So this component wouldn't be able to clear out the region, because it isn't included in the filters this components has access to via the filters prop.

Reply:
Right now we have filters that
- are in the URL and are editable in the UI (most common)
- are not in the URL and can't be edited (Grantee ID in the recipient record)
- are in the URL and can't be edited (region in the Recipient Record)

This is the case I'm handling above. The filters in the parent component appear in the URL and in the component below. My solution was to create a prop (probably poorly named) to hold such filters. The other option I thought of would have been to maintain three arrays of filters instead of two. I think if you have a way to make it simpler (or an idea for me to chew on) I'd love to hear it

Reply:
> are in the URL and are editable in the UI (most common)
are not in the URL and can't be edited (Grantee ID in the recipient record)
are in the URL and can't be edited (region in the Recipient Record)

Thinking out loud, but of these three it seems to me only the first should be passed to this component. `granteeId` should be able to be added into the parent of the `FilterMenu` whenever needed (when the query string is being built to send off to the API). I think `regionId` can be handled the same way as `granteeId`, in the parent component right before the request is sent off.

I think the last point may bring up an issue with how we are structuring URLs on the frontend. I think the case could be made that having the region as a query param isn't appropriate and it should be a part of the path. Especially since it is required to load the resource. It would make sense to me to give the user a 404 if they try to view the recipient tta record page with an invalid `regionId`, one that the recipient has no grants in.

_Might_ be a good rule to have, only change-able filters show up in the filter menu/pills and only change-able filters show up as url params.

Reply:
>  think the last point may bring up an issue with how we are structuring URLs on the frontend. I think the case could be made that having the region as a query param isn't appropriate and it should be a part of the path. Especially since it is required to load the resource. It would make sense to me to give the user a 404 if they try to view the recipient tta record page with an invalid regionId, one that the recipient has no grants in.

>Might be a good rule to have, only change-able filters show up in the filter menu/pills and only change-able filters show up as url params.

That makes sense. We're devoting a lot of energy in the code to partitioning things off in this manner. A lot of props and filtering would go away if we just did as you said - all filters that are query params get passed into the filter menu, and all filters in the menu can be changed. 

Everything else needs to be in the path or not in the URL at all and can't be changed.